site/docs/guides/schema-evolution.md

```diff
@@ -0,0 +1,244 @@
+
+materializations:
+  acmeCo/data-warehouse/materialize-snowflake:
+    endpoint:
+      connector:
+        image: ghcr.io/estuary/materialize-snowflake:v1
+        config: encrypted-snowflake-config.sops.yaml
+    bindings:
+      - source: acmeCo/inventory/anvils_v2
+        resource:
+          table: anvils_v2
+          schema: inventory
+```
+
+The existing `acmeCo/inventory/anvils` collection will not be modified and will remain in place, but won't update because no captures are writing to it.
+
+Note that the collection is now being materialized into a new Snowflake table, `anvils_v2`. This is because the primary key of the `anvils` table doesn't match the new collection key. New data going forward will be added to `anvils_v2` in the data warehouse.
+
+:::warning
+Currently, changing the `target` collection in the capture spec will _not_ cause the capture to perform another backfill. This means that the `anvils_v2` table will get all of the _new_ data going forward, but will not contain the existing data from `anvils`.
+
+We will soon release updates that make it much easier to keep your destination tables fully in sync without needing to change the names. In the meantime, feel free to [reach out on Slack](https://join.slack.com/t/gazette-dev/shared_invite/enQtNjQxMzgyNTEzNzk1LTU0ZjZlZmY5ODdkOTEzZDQzZWU5OTk3ZTgyNjY1ZDE1M2U1ZTViMWQxMThiMjU1N2MwOTlhMmVjYjEzMjEwMGQ) for help.
+:::
+
+**Auto-Discovers:**
+
+If you enabled the option to [**Automatically keep schemas up to date** (`autoDiscover`)](../concepts/captures.md#automatically-update-captures) and selected **Breaking change re-versions collections** (`evolveIncompatibleCollections`) for the capture, this evolution would be performed automatically.
+
+### A new field is added
+
+*Scenario: this is one way in which the schema can change.*
+
+When a new field appears in the collection schema, it _may_ automatically be added to any materializations that use `recommended` fields. Recommended fields are enabled by default in each binding. See [the materialization docs](../concepts/materialization.md#projected-fields) for more info about how to enable or disable `recommended` fields.
+
+When recommended fields are enabled, new fields are added automatically if they meet the criteria for the particular materialization connector. For example, scalar fields (strings, numbers, and booleans) are considered "recommended" fields when materializing to database tables.
+
+If your materialization binding is set to `recommended: false`, or if the new field is not recommended, you can manually add it to the materialization.
+
+To manually add a field:
+
+* **In the Flow web app,** [edit the materialization](./edit-data-flows.md#edit-a-materialization), find the affected binding, and click **Show Fields**.
+* **Using flowctl,** add the field to `fields.include` in the materialization specification as shown [here](../concepts/materialization.md#projected-fields).
+
+### A field's data type has changed
```

Comment:
@psFried would the UI show that schema evolution dialog option (like with the changed key & projections) in this case?

Reply:
Technically, it depends on whether or not the connector permits the change. If the connector says it's OK, then we won't show any errors or an evolution dialog. But if the connector rejects the change, then we will. It can be a little nuanced, but some general guidelines are:

- materializations to S3 or webhooks will generally tolerate most any change
- database materializations will generally reject most changes to materialized columns, except for dropping `NOT NULL` constraints
- If a given field isn't part of the field selection, then any change will be allowed by the connector (even if it's not a good idea for other reasons)

But yeah, probably the most common scenario would be that changing a field type would result in the UI showing the schema evolution dialog.